src/wasm/cpp/README.md

```diff
@@ -2,36 +2,47 @@
 
 ## Setup
 
-This is only required once per checkout.
+Our C++ code is built by Bazel. You will need to install Bazel, and use it to
+build and run the wasm code/tests (everything under `src/wasm/`).
 
-- `tools/sigh install emsdk-npm`
-
-Alternatively, pass `--install` to the `wasm` command the first time you use it.
+1. **Installing Bazel:** run the `tools/setup` script to install Bazel on Linux
+   (or otherwise look at that script to see which version of Bazel to download
+   and how to set it up).
 
 ## Build
 
 See [here](particles/WasmTemplate) for a working example.
 
-- Define a manifest file with the particle spec referencing a `.wasm` module
+- Add a `BUILD` file in the relevant directory
+- Generate a C++ header file from your particle spec using the `arcs_cc_schema`
+  build rule:
+  ```
+  arcs_cc_schema(
+      name = "example_schema",
+      src = "example.arcs",
+      out = "example.h",
+  )
+  ```
 - Write your C++ particle(s)
-  - #include <arcs.h> and the schema2packager-generated header(s)
-  - Particles inherit from arcs::Particle and use arcs::Singleton and arcs::Collection
-- Create a `wasm.json` file to set up the sigh-based build:
-
-  ```json5
-  {
-    "module.wasm": {               // output filename for the compiled wasm module
-      "manifest": "example.arcs",  // manifest file; defines the recipe, particle spec and schemas
-      "genfile": "schemas.h",      // name of file to be generated by schema2packager
-      "src": ["example.cc", ...],  // source files
-      "outdir": "$here"            // location of wasm module and schema2packager output
-    }
-  }
+  - Include the Arcs C++ header: `#include "src/wasm/cpp/arcs.h"`.
+  - Include your generated schema header(s): `#include "particles/WasmTemplate/example.h"`.
+  - Particles inherit from `arcs::Particle` and use `arcs::Singleton` and
+    `arcs::Collection`
+- Add a `wasm_cc_binary` build rule for your C++ particle(s) to your `BUILD`
+  file:
+  ```
+  cc_wasm_binary(
+      name = "example_particle",
+      srcs = ["example.cc"],
+      hdrs = ["example.h"],
```

Comment:
How do these fields relate to the wasm.json ones? Does `name` dictate the wasm binary, and `hdrs` the generated class file? It'd be useful to keep the inline comments (updated as necessary) for clarity.

Reply:
Added inline comments